python/taichi/lang/_ndarray.py

```diff
@@ -307,11 +307,15 @@ def __init__(self, arr, indices_first, indices_second):
             self.indices = indices_first + indices_second
 
         def getter():
+            if impl.get_runtime().prog.config.arch == _ti_core.cuda:
```

Comment:
Nice work!! Hmm I wonder if it's possible to move these into C++ where we need the context? (that way it's more precise and less error prone IMHO

Reply:
Yeah this is an ad-hoc patch for ndarray, But it is not easy to move inner as we can't casually set and pop out the CUDA context. It doesn't take effect even if I place this inside the accessor bank's setter and getter methods. So for now I have to put it here embarrassingly